Generator.Equals/Models/EqualityTypeModelTransformer.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Generator.Equals.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Generator.Equals.Models;
+
+internal sealed class EqualityTypeModelTransformer
+{
+    private readonly GeneratorAttributeSyntaxContext _context;
+    private readonly CancellationToken _token;
+
+    public EqualityTypeModelTransformer(GeneratorAttributeSyntaxContext context, CancellationToken token)
+    {
+        _context = context;
+        _token = token;
+    }
+
+    public EqualityTypeModel Transform()
+    {
+        var attributesMetadata = AttributesMetadata.Instance;
+
+        var equatableAttributeData = _context.Attributes.SingleOrDefault();
+        if (equatableAttributeData is null || !attributesMetadata.Equatable.Equals(equatableAttributeData.AttributeClass))
+        {
+            throw new Exception("Expected exactly one EquatableAttribute.");
+        }
+
+
+        var explicitMode = equatableAttributeData.GetNamedArgumentValue("Explicit") is true;
+        var ignoreInheritedMembers = equatableAttributeData.GetNamedArgumentValue("IgnoreInheritedMembers") is true;
+
+        if (_context.TargetSymbol is not ITypeSymbol symbol)
+        {
+            throw new Exception("Expected a type symbol.");
+        }
+
+        var baseTypeName = symbol.BaseType?.ToFQF();
+
+        // Extract the symbol's fully qualified name
+        var typeName = symbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat);
+
+        var fullname = symbol.ToDisplayString();
+
+        var containingSymbols = GetContainingSymbols(symbol, includeSelf: symbol.IsRecord);
+        var bems = EqualityMemberModelTransformer.BuildEqualityModels(symbol, attributesMetadata, explicitMode);
+
+        var model = new EqualityTypeModel(typeName, containingSymbols, attributesMetadata, explicitMode, ignoreInheritedMembers)
+        {
+            BuildEqualityModels = bems,
+            IsSealed = symbol.IsSealed,
+            BaseTypeName = baseTypeName,
+            Fullname = fullname,
+            SyntaxKind = _context.TargetNode.Kind()
+        };
+
+        if (model.SyntaxKind is not (
+            SyntaxKind.StructDeclaration
+            or SyntaxKind.RecordStructDeclaration
+            or SyntaxKind.RecordDeclaration
+            or SyntaxKind.ClassDeclaration
+            )
+           )
+        {
+            throw new Exception("Expected a struct, record struct, record, or class declaration syntax.");
+        }
+
+
+        return model;
+    }
+
+    public static ImmutableArray<ContainingSymbol> GetContainingSymbols(ISymbol symbol, bool includeSelf = false)
+    {
+        var parentSymbols = symbol
+            .GetParentSymbols(includeSelf)
+            .TakeUntilNamespace();
+
+        return parentSymbols.Select(x =>
+            {
+                if (x.IsNamespace)
```

Comment:
`TakeUntilNamespace` implies that this will never be true. You defined the extension method, so maybe it's not, but that's what the naming implies.

Reply:
How about ``TakeUntilAfterNamespace`` ?